mantidproject/mantid
# Add temporary norm and data workspaces to MDNorm*

Description of work.

This adds optional input MDHistoWorkspaces to `MDNormSCD` and `MDNormDirectSC` for accumulating data and normalization from multiple runs. This avoids allocating and deallocating temporary workspaces.

Looking at a user's reduction script on CORELLI (10 runs with 8 symmetry operations), the runtime is reduced by over 3x versus nightly. Compared to v3.9.2, runtimes were reduced by over 13x. On a separate computer, the runtime is reduced by over 7x versus nightly and 53x versus v3.9.2. 

### topaz2.sns.gov (2 x Intel Xeon E5-2670 v3) 

this branch:
real    9m29.327s
user    35m15.398s
sys     8m54.461s

nightly:
real    31m1.981s
user    47m37.292s
sys    22m12.430s (

v3.9.2:
real    127m34.143s
user    2950m20.871s
sys     40m26.429s

### topaz.sns.gov (4 x AMD Opteron 6376)

this branch:
real    17m59.921s
user    53m40.854s
sys     32m7.057s

nightly:
real    132m51.800s
user    127m27.238s
sys     80m5.725s

(1 run, 8 symmetry operations)
real    96m10.263s
user    3677m8.046s
sys     66m44.554s

**To test:**



There is no GitHub issue associated with this pull request

* Review changes. Are variable names and descriptions clear? Are additional runtime checks necessary?
* Update script to use these optional input workspaces (I can provide an example). Verify results match and performance is improved. 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
